core/src/components/datetime/datetime.tsx

```diff
@@ -492,15 +492,24 @@ export class Datetime implements ComponentInterface {
    */
   @Method()
   async confirm(closeOverlay = false) {
-    const { isCalendarPicker, activeParts } = this;
+    const { isCalendarPicker, activeParts, preferWheel, workingParts } = this;
```

Comment:
The wrong date appears to be getting selected when I test the dev build in the linked issue's repro case:

https://github.com/ionic-team/ionic-framework/assets/2721089/9f9a5a1d-a792-4b3f-acd5-b59ad1d165a8

Reply:
Great catch - I think you uncovered another bug 👀 

In their example when setting the working parts to "December 30", `setWorkingParts` gets called twice with "December 29" and then "December 30" - which overwrites your changes.

Reply:
Updated dev-build: `7.5.5-dev.11700511127.1168d13f`. 

Tested against their reproduction (you'll either have to `--force` or `--legacy-peer-deps` during install).